src/collapse/vsprof.rs

```diff
@@ -0,0 +1,188 @@
+use std::{
+    cmp::Ordering,
+    io::{self, BufRead},
+};
+
+use log::warn;
+
+use crate::collapse::{common::Occurrences, Collapse};
+
+static START_LINE: &str = "Level,Function Name,Number of Calls,Elapsed Inclusive Time %,Elapsed Exclusive Time %,Avg Elapsed Inclusive Time,Avg Elapsed Exclusive Time,Module Name,";
+
+/// A stack collapser for the output of the built in Visual Studio profiler.
+#[derive(Default)]
+pub struct Folder {
+    /// Function entries on the stack in this entry thus far.
+    stack: Vec<(String, usize)>,
+}
+
+impl Collapse for Folder {
+    fn collapse<R, W>(&mut self, mut reader: R, writer: W) -> io::Result<()>
+    where
+        R: std::io::BufRead,
+        W: std::io::Write,
+    {
+        // Skip the header
+        let mut line = Vec::new();
+        if reader.read_until(b'\n', &mut line)? == 0 {
+            warn!("File ended before start of call graph");
+            return Ok(());
+        };
+
+        // Process the data
+        let mut occurences = Occurrences::new(1);
+        loop {
+            line.clear();
+            if reader.read_until(b'\n', &mut line)? == 0 {
+                break;
+            }
+            let l = String::from_utf8_lossy(&line);
+            let line = l.trim_end();
+            if line.is_empty() {
+                continue;
+            } else {
+                self.on_line(line, &mut occurences)?;
+            }
+        }
+
+        // Write the results
+        occurences.write_and_clear(writer)?;
+
+        // Reset the state
+        self.stack.clear();
+        Ok(())
+    }
+
+    fn is_applicable(&mut self, input: &str) -> Option<bool> {
+        let mut input = input.as_bytes();
+        let mut line = String::new();
+        if let Ok(n) = input.read_line(&mut line) {
+            if n == 0 {
+                return Some(false);
+            }
+        } else {
+            return Some(false);
+        }
+
+        // Some files may start with the <U+FEFF> character (zero width no-break space). This
+        // causes the call to `starts_with` to return false, which in this case isn't what we want.
+        // As this character has no influence on the rest of the file, we can safely ignore it.
+        if line.starts_with(START_LINE) || line.starts_with(&("\u{feff}".to_owned() + START_LINE)) {
```

Comment:
This saves you the extra string allocation, and I think reads better to:
```suggestion
        if line.trim_start_matches('\u{feff}') == START_LINE {
```